### PR TITLE
CP-12157: Create XenServerHealthCheck framework

### PR DESCRIPTION
**Pull request by cheng--zhang**
1.Create a service named XenServerHealthCheck
2.Read the XenCenter Server List from configuration
3.Try to connect to these servers
4.Modify protection of XenCenter config from CurrentUset to LocalMachine

Signed-off-by: Cheng Zhang <cheng.zhang@citrix.com>

### CODE REVIEW

File: XenCenterLib/EncryptionUtils.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ public static string Protect(string data)
         {
             byte[] dataBytes = Encoding.Unicode.GetBytes(data);
 
-            byte[] protectedBytes = ProtectedData.Protect(dataBytes, GetSalt(), DataProtectionScope.CurrentUser);
+            byte[] protectedBytes = ProtectedData.Protect(dataBytes, GetSalt(), DataProtectionScope.LocalMachine);
```
Comment (GaborApatiNagy) [inline XenCenterLib/EncryptionUtils.cs]: This would change how XenCenter stores data.
While I see why this change was made, this would require users to know that they will need to trust not only administrators, but all the local users on a machine they are using.

Comment (cheng--zhang) [inline XenCenterLib/EncryptionUtils.cs]: Yes, without the scope change, we cannot read the config from a service. because service runing in system account.

Comment (stephen-turner) [inline XenCenterLib/EncryptionUtils.cs]: We couldn't change it without an upgrade path from the CurrentUser encryption. But actually, I think this may not be necessary in the end.

Comment (MihaelaStoica) [inline XenCenterLib/EncryptionUtils.cs]: We are currently investigating other ways of passing the server list to the service, without having to modufy the encryption level
For the time being, we can keep these changes, so that we can progress with the implementation. But we plan to revert them later when a different solution is agreed.
Cheng, could you please include this change into a #if directive, so that the change is only in place when compiling for this testing configuration.

**Comment by cheng--zhang** [comment]
Hi @MihaelaStoica 
Please help to review the code again, I add a parameter called: CALL_HOME_DEV there.

**Comment by MihaelaStoica** [comment]
This looks good Cheng. Just one note, the timer interval is set to 10000 (10 seconds). I think it should be 30*60000 (30 minutes).

**Comment by cheng--zhang** [comment]
Hi @MihaelaStoica 
Sorry about the wrong time. I use it for my test, and just update it, please help to check and merge it if possible.
Thanks

**Comment by MihaelaStoica** [comment]
Looks good. Merging.
